app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -84,11 +94,37 @@ class AuditController @Inject() (
       conversionService.toJson(auditType.messageType.get, request.body)
     else EitherT.rightT(request.body)
 
-  private def getSource(auditType: AuditType, uri: Option[ObjectStoreResourceLocation], request: Request[Source[ByteString, _]])(implicit
+  private def fileId() =
+    s"{UUID.randomUUID().toString}-${DateTimeFormatter.ofPattern("yyyyMMdd-HHmmss.SSS").withZone(ZoneOffset.UTC).format(Instant.now())}"
+
+  private def getSource(auditType: AuditType, uri: Option[ObjectStoreResourceLocation], request: Request[Source[ByteString, _]])(exceedsLimit: Boolean)(implicit
     hc: HeaderCarrier
-  ): EitherT[Future, PresentationError, Source[ByteString, _]] =
-    uri match {
-      case None           => convertIfNecessary(auditType, request).asPresentation
-      case Some(location) => objectStoreService.getContents(location).asPresentation
+  ): EitherT[Future, PresentationError, Payload] =
+    (uri, exceedsLimit) match {
+      case (None, false) => // payload in body and < auditing message limit
+        convertIfNecessary(auditType, request).asPresentation
+          .map(
+            source => Right(source): Payload
+          )
+      case (Some(uri), false) => // payload in object store and < auditing message limit
+        objectStoreService
+          .getContents(uri)
+          .asPresentation
+          .map(
+            source => Right(source): Payload
+          )
+      case (None, true) => // payload in body and too big for auditing so add to object store
```

Comment:
@drnaylor-oc  is this requirement to store file at object store /auditing location in case of size exceeds .5 MB?

Reply:
yes